native/dgemm.go

```diff
@@ -336,54 +247,31 @@ func dgemmSerialTransNot(a, b, c general64, alpha float64) {
 }
 
 // dgemmSerial where neither a is not transposed and b is
-func dgemmSerialNotTrans(a, b, c general64, alpha float64) {
-	if debug {
-		if a.cols != b.cols {
-			panic("inner dimension mismatch")
-		}
-		if a.rows != c.rows {
-			panic("outer dimension mismatch")
-		}
-		if b.rows != c.cols {
-			panic("outer dimension mismatch")
-		}
-	}
-
+//func dgemmSerialNotTrans(a, b, c general64, alpha float64) {
+func dgemmSerialNotTrans(m, n, k int, a []float64, lda int, b []float64, ldb int, c []float64, ldc int, alpha float64) {
 	// This style is used instead of the literal [i*stride +j]) is used because
 	// approximately 5 times faster as of go 1.3.
-	for i := 0; i < a.rows; i++ {
-		atmp := a.data[i*a.stride : i*a.stride+a.cols]
-		ctmp := c.data[i*c.stride : i*c.stride+c.cols]
-		for j := 0; j < b.rows; j++ {
-			ctmp[j] += alpha * asm.DdotUnitary(atmp, b.data[j*b.stride:j*b.stride+b.cols])
+	for i := 0; i < m; i++ {
+		atmp := a[i*lda : i*lda+k]
+		ctmp := c[i*ldc : i*ldc+n]
+		for j := 0; j < n; j++ {
+			ctmp[j] += alpha * asm.DdotUnitary(atmp, b[j*ldb:j*ldb+k])
 		}
 	}
-
 }
 
 // dgemmSerial where both are transposed
-func dgemmSerialTransTrans(a, b, c general64, alpha float64) {
-	if debug {
-		if a.rows != b.cols {
-			panic("inner dimension mismatch")
-		}
-		if a.cols != c.rows {
-			panic("outer dimension mismatch")
-		}
-		if b.rows != c.cols {
-			panic("outer dimension mismatch")
-		}
-	}
-
+//func dgemmSerialTransTrans(a, b, c general64, alpha float64) {
+func dgemmSerialTransTrans(m, n, k int, a []float64, lda int, b []float64, ldb int, c []float64, ldc int, alpha float64) {
 	// This style is used instead of the literal [i*stride +j]) is used because
 	// approximately 5 times faster as of go 1.3.
-	for l := 0; l < a.rows; l++ {
-		for i, v := range a.data[l*a.stride : l*a.stride+a.cols] {
-			ctmp := c.data[i*c.stride : i*c.stride+c.cols]
+	for l := 0; l < k; l++ {
+		for i, v := range a[l*lda : l*lda+m] {
+			ctmp := c[i*ldc : i*ldc+n]
 			if v != 0 {
```

Comment:
This check seems redundant having the one below (in general we assume that `alpha` will be non-zero I guess).